packages/ui/src/components/sha-badge.tsx

```diff
@@ -44,13 +44,20 @@ function Icon({ ...props }: IconProps) {
   const { children, handleClick } = props
 
   return (
-    <button
-      className="flex items-center border-l bg-background px-1.5 py-0.5 hover:bg-muted/50"
+    <div
+      role="button"
```

Comment:
Instead of trying to get a `div` to act like a `button`, let's use an actual `button`